src/lib/consensus/proof_of_stake.ml

```diff
@@ -3916,6 +3914,711 @@ let%test_module "Proof of stake tests" =
       let within_tolerance = diff < tolerance in
       if not within_tolerance then
         failwithf "actual vs. expected: %d vs %f" actual expected ()
+
+    (* Consensus selection tests. *)
+
+    let sum_lengths = List.fold ~init:Length.zero ~f:Length.add
+
+    let gen_pair a b =
+      let open Quickcheck.Generator.Let_syntax in
+      let%map xa = a and xb = b in
+      (xa, xb)
+
+    let gen_triple a b c =
+      let open Quickcheck.Generator.Let_syntax in
+      let%map xa = a and xb = b and xc = c in
+      (xa, xb, xc)
+
+    let rec gen_exlusion ls ~gen ~equal =
+      let open Quickcheck.Generator.Let_syntax in
+      let%bind x = gen in
+      if List.mem ls x ~equal then gen_exlusion ls ~gen ~equal else return x
+
+    let gen_unique_list amount ~gen ~equal =
```

Comment:
Yeah, I'm aware of that. I'll put a comment out explicitly. Probably doesn't matter for here since the largest size is 8, but is there a better way to do this with quickcheck?